lib/poseidon/sync_producer.rb

```diff
@@ -42,7 +42,12 @@ def send_messages(messages)
       messages_to_send = MessagesToSend.new(messages, @cluster_metadata)
 
       if refresh_interval_elapsed?
-        refresh_metadata(messages_to_send.topic_set)
+        begin
+          refresh_metadata(messages_to_send.topic_set)
+        rescue Errors::UnableToFetchMetadata => e
```

Comment:
I've verified that this is the only error we see in production and also having followed the code paths.  What do you think about increasing the next refresh time by some number of seconds (e.g. 10s) to avoid hitting the error on every single produce.

Granted, I don't think this is a real situation we're going to run into (where we constantly are unable to get metadata), but I'm happy to put something in place here.

I'm not sure I follow how it'll affect the accuracy of the messages.  Kafka will fail `produce` calls if we attempt to send messages to a broker that's not the leader for that partition.